message.js

```diff
@@ -149,22 +146,12 @@ export default class MessageCall  {
     delete this._callsWaiting[message.id];
     assert(callWaiting, "Got a response for call ID " + message.id + ", but we did not make such a call, or we already got the response for it");
     if (message.success) {
-      try {
-        callWaiting.resolve(message.result);
-      } catch (ex) {
-        console.error(ex); // TODO remove
-        callWaiting.reject(ex); // TODO can I call reject() after resolve()?
-      }
+      callWaiting.resolve(message.result);
     } else {
       let ex = new Error();
       ex.message = message.message;
       ex.code = message.code;
-      try {
-        callWaiting.reject(ex);
-      } catch (ex) {
-        console.error("Our error handling threw");
-        console.error(ex);
-      }
+      callWaiting.reject(ex);
```

Comment:
Why did you remove this safety net?

Reply:
What are you trying to protect against here? `reject` never returns or throws anything.

Reply:
I thought that reject() would create a new call stack, which throws the exception, lets it bubble, it will be caught somewhere, and that catch code might display the error in the UI, or something like that. If that error handling code fails and throws an exception, who handles that?

Reply:
But maybe I'm misunderstanding it and an exception thrown in the error handling won't end up here. But where does it end up, then?